### ISSUE TITLE
Add hyper-parameter search on the server

### ISSUE THREAD

**Issue by Nacho114** [epfml/DeAI#151]
Suggested by @Saipraneet on [here](https://github.com/epfml/FeAI/issues/3)

Can we add some functionality at the server level to do hyper-parameter search as well? This I think is quite useful and necessary.

Essentially, the server can start training with different hyperparameters on different clients, and keep track of this. We can use this to for e.g. figure out what learning rate to use etc.
